examples/main.go

```diff
@@ -0,0 +1,249 @@
+	entries, err := fs.ReadDir(root)
+	if err != nil {
+		return nil, err
+	}
+
+	files := map[string]string{}
+	for _, entry := range entries {
+		path := filepath.Join(root, entry.Name())
+
+		if entry.IsDir() {
+			e.dirs = append(e.dirs, path)
+			continue
+		} else if !strings.HasSuffix(entry.Name(), ".yaml") && !strings.HasSuffix(entry.Name(), ".yml") {
+			continue
+		}
+
+		content, err := fs.ReadFile(path)
+		if err != nil {
+			return nil, err
+		}
+
+		// Parent path as key and content as value
+		path = keyFn(root)
+		files[path] = joinYamls(files[path], string(content))
+	}
+
+	return files, nil
+}
+
+// joinYamls takes a list of yaml strings and joins them into a single yaml
+func joinYamls(yamls ...string) string {
+	joined := ""
+	for _, yaml := range yamls {
+		if yaml == "" {
+			continue
+		}
+
+		joined += "\n---\n" + yaml
+	}
+
+	return joined
+}
+
+// ClusterClassRegex returns ClusterClasses matching the given search regex.
+func (e *Examples) ClusterClassRegex(search *regexp.Regexp) (string, error) {
+	keys := []string{}
+	for k := range e.ClusterClasses {
+		if search.MatchString(k) {
+			keys = append(keys, k)
+		}
+	}
+
+	if len(keys) == 0 {
+		return "", fmt.Errorf("clusterclasses not found: %s, maybe you want to use: %s?\n", search, e.closestMatch(search.String()))
+	} else {
+		data := slices.Collect(maps.Values(e.Applications))
+		for _, key := range keys {
+			data = append(data, e.ClusterClasses[key])
+		}
+
+		return joinYamls(data...), nil
+	}
+}
+
+// ClusterClass returns the ClusterClass specified by the given key.
+func (e *Examples) ClusterClass(key string) (string, error) {
+	if clusterClass, found := e.ClusterClasses[key]; !found {
+		return "", fmt.Errorf("clusterclasses not found: %s, maybe you want to use: %s?\n", key, e.closestMatch(key))
+	} else {
+		data := slices.Collect(maps.Values(e.Applications))
+		data = append(data, clusterClass)
+
+		return joinYamls(data...), nil
+	}
+}
+
+// closestMatch finds the closest matching key from the ClusterClasses map
+
+func (e *Examples) closestMatch(key string) string {
+	closestMatch, shortestDistance := "", math.MaxInt
+	for k := range e.ClusterClasses {
+		distance := levenshtein.ComputeDistance(k, key)
+		if distance < shortestDistance {
+			shortestDistance = distance
+			closestMatch = k
+		}
+	}
+
+	return closestMatch
+}
+
+// initFlags initializes the flags.
+func initFlags(fs *pflag.FlagSet) {
+	fs.BoolVarP(&clusterClassList, "list", "l", false,
+		"List cluster class names from examples")
+
+	fs.StringVarP(&clusterClassRegex, "regex", "r", "",
+		"ClusterClass search regex")
+}
+
+func main() {
+	initFlags(pflag.CommandLine)
+	pflag.CommandLine.AddGoFlagSet(flag.CommandLine)
+	pflag.Parse()
+
+	examples := Examples{}
+	if err := examples.Collect(); err != nil {
+		fmt.Fprintf(os.Stderr, "%v", err)
+
+		os.Exit(1)
+	}
+
+	if clusterClassList {
+		keys := maps.Keys(examples.ClusterClasses)
+		fmt.Printf("Available classes: %s", slices.Collect(keys))
```

Comment:
```suggestion
		fmt.Printf("Available classes: %s\n", slices.Collect(keys))
```
Nicer with a new line at the end I think.

Reply:
Done